odoo/addons/test_lint/tests/_odoo_checker_sql_injection.py

```diff
@@ -65,7 +83,6 @@ def _check_concatenation(self, node):
         # ignore sql.SQL().format
         if isinstance(node, astroid.Call) \
                 and isinstance(node.func, astroid.Attribute) \
-                and isinstance(node.func.expr, astroid.Const) \
```

Comment:
Looking at the CI failure and thinking about it more, rather than put the psycopg2.sql check in `_allowable` it should probably be here whitelisting an expr inferred as an sql structure, rather than blacklisting a const (which as you've demonstrated is not sufficient).